src/storage/src/statistics.rs

```diff
@@ -951,22 +941,30 @@ impl AggregatedStatistics {
         self.global_sink_statistics.remove(&id);
     }
 
-    /// Advance the _global epoch_ for statistics. Local statistics before this epoch will be
-    /// ignored.
+    /// Advance the _global epoch_ for statistics.
+    ///
+    /// Gauge values from previous epochs will be ignored. Counter values from previous epochs will
+    /// still be applied as usual.
     pub fn advance_global_epoch(&mut self, id: GlobalId) {
-        if self.worker_id == 0 {
-            self.global_source_statistics
-                .entry(id)
-                .and_modify(|(ref mut epoch, ref mut stats)| {
-                    *epoch += 1;
-                    *stats = vec![None; self.worker_count];
-                });
-            self.global_sink_statistics
-                .entry(id)
-                .and_modify(|(ref mut epoch, ref mut stats)| {
-                    *epoch += 1;
-                    *stats = vec![None; self.worker_count];
-                });
+        if self.worker_id != 0 {
```

Comment:
Just for my education, why don't we reset the other workers?

Reply:
Only worker 0 maintains global statistics, so there shouldn't be any resetting necessary for other workers. It also wouldn't hurt to still do it. I'll remove this guard.